allensdk/brain_observatory/filtered_running_speed/filtered_running_speed.py

```diff
@@ -0,0 +1,382 @@
+        behavior_velocities: pandas.core.frame.DataFrame
+           Velocity data from behavior stimulus
+        replay_velocities: pandas.core.frame.DataFrame
+            Velocity data from replay stimulus
+        frame_times: numpy.ndarray
+            list of the vsync times
+
+        Returns
+        -------
+        list[pd.DataFrame, pd.DataFrame]
+            concatenated velocity data, raw data
+        """
+
+        speed = np.concatenate(
+            (
+                behavior_velocities['speed'],
+                mapping_velocities['speed'],
+                replay_velocities['speed']),
+            axis=None
+        )
+
+        dx = np.concatenate(
+            (
+                behavior_velocities['dx'],
+                mapping_velocities['dx'],
+                replay_velocities['dx']),
+            axis=None
+        )
+
+        vsig = np.concatenate(
+            (
+                behavior_velocities['v_sig'],
+                mapping_velocities['v_sig'],
+                replay_velocities['v_sig']),
+            axis=None
+        )
+
+        vin = np.concatenate(
+            (
+                behavior_velocities['v_in'],
+                mapping_velocities['v_in'],
+                replay_velocities['v_in']),
+            axis=None
+        )
+
+        frame_indexes = list(
+            range(FilteredRunningSpeed.START_FRAME, len(frame_times))
+        )
+
+        velocities = pd.DataFrame(
+            {
+                "velocity": speed,
+                "net_rotation": dx,
+                "frame_indexes": frame_indexes,
+                "frame_time": frame_times
+            }
+        )
+
+        # Warning - the 'iscose' line below needs to be refactored
+        # is it exists in multiple places
+
+        # due to an acquisition bug (the buffer of raw orientations
+        # may be updated more slowly than it is read, leading to
+        # a 0 value for the change in orientation over an interval)
+        # there may be exact zeros in the velocity.
+        velocities = velocities[~(np.isclose(velocities["net_rotation"], 0.0))]
+
+        raw_data = pd.DataFrame(
+            {"vsig": vsig, "vin": vin, "frame_time": frame_times, "dx": dx}
+        )
+
+        return velocities, raw_data
+
+    def _process_single_simulus_experiment(
+        self
+    ):
+        """
+        Process an experiment with a single simulus sessions
+        """
+
+        start_index = FilteredRunningSpeed.START_FRAME
+
+        frame_times = self._get_frame_times()
+
+        end_index = len(frame_times)
+
+        velocities, raw_data = self._extract_dx_info(
+            frame_times,
+            start_index,
+            end_index,
+            self.pkl_path
+        )
+
+        store = pd.HDFStore(self.args['output_path'])
+        store.put("running_speed", velocities)
+        store.put("raw_data", raw_data)
+        store.close()
+
+        self._write_output_file()
+
+    def _write_output_file(self):
```

Comment:
Can you rename this `_write_output_json`? I found it confusing that the output HDF5 file is actually written elsewhere.

Reply:
I've updated these.